File: apis/events/serverless/func.js
const fdk=require('@fnproject/fdk');
const axios=require('axios');

let access_token = '';

let ORDS_HOSTNAME = '';
let APEX_WORKSPACE = '';
let API_USER = '';
let API_PASSWORD = '';

const token_refresh_interval = 1800000;

function getToken() {
    console.log('retrieve token');
    if (ORDS_HOSTNAME != '') {
        axios.post('https://'+ORDS_HOSTNAME+'/ords/'+APEX_WORKSPACE+'/oauth/token', 'grant_type=client_credentials', { auth: { username: API_USER, password: API_PASSWORD }})
        .then(oauthres => {
                  access_token = oauthres.data.access_token;
	          console.log('token:'+access_token);
              })
        .catch(err => {
                  console.log(err);
                  exit(-1);
              });
    }
}

getToken();
setInterval(getToken, token_refresh_interval);

async function addEvent(input,ctx) {
    ORDS_HOSTNAME = ctx.config['ORDS_HOSTNAME'];
    APEX_WORKSPACE= ctx.config['APEX_WORKSPACE'];
    API_USER = ctx.config['API_USER'];
    API_PASSWORD= ctx.config['API_PASSWORD'];
    getToken();
    adwres = await axios.post('https://'+ORDS_HOSTNAME+'/ords/'+APEX_WORKSPACE+'/event_table/', input, { headers: { 'Authorization': 'Bearer '+access_token }});
    return adwres.data;
}

fdk.handle(function(input,ctx){
  try {
    return addEvent(input,ctx);
  } catch (error) {
    console.log(error);
    return { 'response' : 'bad' };
  }
})

File: apis/events/serverless/python/func.py
import io
import json
import logging
import requests
from requests.auth import HTTPBasicAuth

from fdk import response

def handler(ctx, data: io.BytesIO = None):
    try:
        ORDS_HOSTNAME = ctx.Config()['ORDS_HOSTNAME']
        APEX_WORKSPACE = ctx.Config()['APEX_WORKSPACE']
        API_USER = ctx.Config()['API_USER']
        API_PASSWORD = ctx.Config()['API_PASSWORD']
        body = json.loads(data.getvalue())
        res = requests.post('https://'+ORDS_HOSTNAME+'/ords/'+APEX_WORKSPACE+'/oauth/token', data={'grant_type':'client_credentials'}, auth=HTTPBasicAuth(API_USER, API_PASSWORD));
        access_token = res.json()['access_token'];
        res = requests.post('https://'+ORDS_HOSTNAME+'/ords/'+APEX_WORKSPACE+'/event_table/', body, headers={ 'Authorization': 'Bearer '+access_token});
        return response.Response(ctx, response_data=res.json(),headers={"Content-Type": "application/json"})
    except (Exception, ValueError) as ex:
        logging.getLogger().info('error parsing json payload: ' + str(ex))
        return response.Response(ctx, str(ex))

    logging.getLogger().info("Inside Python Hello World function")
